ftp.txt.em

```diff
@@ -0,0 +1,14 @@
+open @(ip)
+anon
+bin
+prompt
+mput @
+@[for pkg in ws.pkgs]@
+@[if len(pkg.objects) > 0]@
+@[for (src, obj) in pkg.objects]@
+"@(ws.build.path)\@(obj)" @
+@[end for]@
+@[end if]@
+@[end for]@
```

Comment:
If/when changing type of variables, or changing fields of `STRUCTURE` type variables, uploading new `.pc`s which reference those variables has a high chance of failing (see also [this troubleshooting entry](http://wiki.ros.org/fanuc_driver/Troubleshooting#MEMO-159:_Convert_failed_in_PROG) for another Karel-based project).

It would perhaps be good to first `DEL` or `MDEL` to remove all known objects and then upload the new ones.

An issue with that could be that `.vr` would also need to be removed, as those are actually also problematic.

Reply:
good point. updated in 4a3b457. I don't have roboguide on hand right now, so I haven't tested it, but I will post tomorrow if there are any unforeseen problems with what I have done.

Reply:
This is working for me now on roboguide after 3a2f768